libfreerdp/core/security.c

```diff
@@ -524,9 +524,9 @@ static void fips_expand_key_bits(BYTE* in, BYTE* out)
 		p = b / 8;
 		r = b % 8;
 
-		if (r == 0)
+		if (r <= 1)
 		{
-			out[i] = buf[p] & 0xfe;
+			out[i] = (buf[p] << r) & 0xfe;
```

Comment:
Why did you add the shift here?

Reply:
Good question! :)

Reply:
Ok, missed the commit message in overview mode, so this looks good too.

Reply:
In order to handle the case when r=1 (without access into buf[p+1])...

Reply:
Ah - alright - understood!